cmake.deps/cmake/DepsUtil.cmake

```diff
@@ -0,0 +1,12 @@
+function(construct_patch_command VARIABLE PATCH_CMD PATCH_EXE START_DIR PATCH_FILE)
```

Comment:
I'm not sure we want to go this rabbit-hole with cmake's string parsing, how about creating a function that can handle the patching, which can be invoked similar to any other cmake script?

Here's a version based on `z_vcpkg_apply_patches` (simplified error handling), which can handle multiple patches as a bonus :)

```cmake
function(apply_patches)
  cmake_parse_arguments(PARSE_ARGV 0 "arg" "" "SOURCE_PATH" "PATCHES")
  find_program(Git QUIET)
  find_program(Patch QUIET)

  foreach(patch IN LISTS arg_PATCHES)
    message(STATUS "Applying patch ${patch}")
    if(Git_FOUND)
      execute_process(
        COMMAND "${GIT}" -c core.longpaths=true -c core.autocrlf=false --work-tree=. --git-dir=.git apply "${patch}" --ignore-whitespace --whitespace=nowarn --verbose
        ERROR_VARIABLE error
        WORKING_DIRECTORY "${arg_SOURCE_PATH}"
        RESULT_VARIABLE error_code
      )
    elseif(Patch_FOUND AND UNIX)
      execute_process(
        COMMAND patch -d . -i ${PATCH_FILE}
        ERROR_VARIABLE error
        WORKING_DIRECTORY "${arg_SOURCE_PATH}"
        RESULT_VARIABLE error_code
      )
    else()
      message(FATAL_ERROR "Either git or patch is required to apply patches. Aborting...")
    endif()

    if(error_code)
      message(FATAL_ERROR "Applying patch failed: ${error}")
    endif()
  endforeach()
endfunction()

```

Reply:
Hmm, I don't get it. Should we call this function before or after the `ExternalProject_Add`?

Reply:
you pass it as a `PATCH_COMMAND`

https://github.com/neovim/neovim/blob/c54c37ee28cf380199a65c5849b3abe7aaae0afa/cmake.deps/cmake/BuildLibuv.cmake#L21-L24

so it would be 
```cmake
  PATCH_COMMAND
    ${CMAKE_COMMAND} -DSOURCE_PATH="..." -DPATCHES="..." -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/ApplyPatches.cmake

```

Reply:
I've already got the patching to work in my first commit. I'm trying alternatives to see if there's a way to do it without calling the script with `-P`.

Reply:
I'll revert to my first commit and mark it as ready as I indeed don't think cmake's string parsing is gonna work out.